### PR TITLE
Move shell to //flutter and split shell/BUILD.gn into smaller pieces for each subcomponent.

### PR DESCRIPTION
**Pull request by chinmaygarde**
* Namespaces have been updated to reflect the move from //flutter/sky/shell to //flutter/shell.
* shell/BUILD.gn file has been split into smaller GN files for each subcomponent of the shell (common, GPU, diagnostic, testing).
* GN dependencies have been rewritten to stop exposing common shell dependencies as public. Duplicates have also been removed.
* GPU subcomponent has been updated make it more suitable for Vulkan integration.
* The GLFW backend has been resurrected.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @jason-simmons @abarth

**Comment by chinmaygarde** [comment]
This only patches the Mac platform, will patch others before landing.

**Comment by abarth** [comment]
This looks great.  Let me know when you've got all the platforms sorted and I'll take a look in more detail.

**Comment by chinmaygarde** [comment]
Sure. I also realized that platform specific stuff can also be split into its smaller GN files. Attempting that now.

**Comment by chinmaygarde** [comment]
The Mac desktop and iOS targets now have their own GN files. Several times, I have been told that it was rather confusing that "mac" files were being compiled for iOS as well. Since mac platform shares most of these files, the structure of the project has been updated to reflect this commonality. The project now looks like so:
* shell
  * platform
    * mac
      * ios
      * desktop
    * android
    * linux

**Comment by abarth** [comment]
maybe s/mac/darwin/ and s/desktop/mac/ ?

**Comment by chinmaygarde** [comment]
> maybe s/mac/darwin/ and s/desktop/mac/ ?

Ok.

**Comment by eseidelGoogle** [comment]
Somehow you ended up with a funny author on your last commit. :)

**Comment by chinmaygarde** [comment]
Yeah, sorry about that. I am editing files on my mac and pushing them over the uniquity instance to test the Linux targets :( Committed an update on that machine. Will rebase.

**Comment by chinmaygarde** [comment]
All platforms have been fixed and the GLFW backend has been resurrected.

File: shell/platform/linux/platform_view_glfw.cc [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  pointer_data->pressure = 1.0;
+  pointer_data->pressure_max = 1.0;
+
+  auto pointer_packet = pointer::PointerPacket::New();
+  pointer_packet->pointers.push_back(pointer_data.Pass());
+  engine_->OnPointerPacket(pointer_packet.Pass());
+}
+
+void PlatformViewGLFW::OnKeyEvent(int key, int scancode, int action, int mods) {
+#if 0
```
Comment (abarth) [inline shell/platform/linux/platform_view_glfw.cc]: ?

Comment (chinmaygarde) [inline shell/platform/linux/platform_view_glfw.cc]: Will delete. Has to be moved to the platform service provider.

**Comment by chinmaygarde** [comment]
Fixed and rebased. Landing.

**Pull request merged**